apps/web/next.config.js

```diff
@@ -183,7 +183,12 @@ const nextConfig = {
     return config;
   },
   async rewrites() {
-    return [
+    // .* matches / as well(Note: *(i.e wildcard) doesn't match / but .*(i.e. RegExp) does)
+    // It would match /free/30min but not /bookings/upcoming because 'bookings' is an item in pages
+    // It would also not match /free/30min/embed because we are ensuring just two slashes
+    const userTypeRouteRegExp = `/:user((?!${pages.join("|")})[^/]*)/:type`;
```

Comment:
Using [^/]* ensures that we don't match something like /free/30min/embed

Reply:
you could also do `/:user((?!${pages.join("|")})+)/:type` to loose the greediness

Reply:
Did you mean something like .+**?** . I tried that as well even with lazy it would consume all the characters of /free/30min/embed. Because even if lazy it always tries to get a match in the end.